### PR TITLE
Ais 496 refactor

### PR DESCRIPTION
**Pull request by dstuebe**
Refactored AIS to use a separate process for the metadata cache and data set/source subscribers to avoid race conditions in operating on resources. 
Also factored out a number of common methods for use in multiple places by AIS workers.

### CODE REVIEW

File: ion/integration/ais/app_integration_service.py [hunk 1]
```diff
@@ -85,31 +63,70 @@ class AppIntegrationService(ServiceProcess):
...
+        Consider changing to use spanw child - but we need to get the actual instance object not just the id...
+
+        """
+
+        class AIS_Worker_Process(Process, AIS_Mixin):
+            """
+            Worker process for the AIS service
+            """
+
+        worker = AIS_Worker_Process(**{'prcoc-name':name})
```
Comment (daf) [inline ion/integration/ais/app_integration_service.py]: typo - should be proc-name

File: ion/integration/ais/test/test_metadata_cache.py [hunk 2]
```diff
@@ -51,7 +53,7 @@ TEST_RESOURCE_ID = '01234567-8abc-def0-1234-567890123456'
 DISPATCHER_RESOURCE_TYPE = object_utils.create_type_identifier(object_id=7002, version=1)
 
 
-class MetadataCacheTest(IonTestCase):
+class MetadataCacheTest(IonTestCase, AIS_Mixin):
```
Comment (daf) [inline ion/integration/ais/test/test_metadata_cache.py:56]: Hard to test from here, but is this necessary or just a leftover from testing

File: ion/integration/ais/test/test_metadata_cache.py [hunk 3]
```diff
@@ -222,6 +200,8 @@ class MetadataCacheTest(IonTestCase):
 
         dsourcelist = self.cache.getDataSources()
         for ds in dsourcelist:
+
+            print 'DKLS KLSD DSKLD LKS DSLNFLSKDNFIOFHLBLJSKDB   KLSNDKLSN'
```
Comment (daf) [inline ion/integration/ais/test/test_metadata_cache.py]: If you're revising things you can remove this too :)

**Comment by daf** [comment]
Just a few small fixups (typos/leftovers), possible discussion on the test inheriting from AIS_Mixin too but just wanted to make sure that wasn't an artifact.

File: ion/integration/ais/test/test_metadata_cache.py (hunk 2 above)
Comment (dstuebe) [inline ion/integration/ais/test/test_metadata_cache.py:56]: That is required

File: ion/integration/ais/test/test_metadata_cache.py (hunk 3 above)
Comment (dstuebe) [inline ion/integration/ais/test/test_metadata_cache.py]: That one - not so much :-)

**Comment by daf** [comment]
Solid

**Pull request merged**